app/src/main/java/com/keylesspalace/tusky/db/dao/TimelineStatusDao.kt

```diff
@@ -183,6 +183,16 @@ WHERE tuskyAccountId = :tuskyAccountId AND serverId = :statusId"""
     )
     abstract suspend fun setVoted(tuskyAccountId: Long, statusId: String, poll: String)
 
+    @Transaction
+    @OptIn(ExperimentalStdlibApi::class)
+    open suspend fun setShowResults(tuskyAccountId: Long, statusId: String, moshi: Moshi) {
+        getStatus(tuskyAccountId, statusId)?.let { status ->
+            status.poll?.let { poll ->
+                setVoted(tuskyAccountId, statusId, moshi.adapter<Poll>().toJson(poll.copy(voted = true)))
+            }
+        }
+    }
```

Comment:
I don't *love* this, but it feels better than adding another field and updating the "which poll view should I show" logic 🤷

Reply:
By *this* you mean setting the poll to voted when it isn't actually voted? I think that is fine.

But you could include the poll or the whole status in the `PollShowResultsEvent` then you don't need the `getStatus` here.

Reply:
By `this` I mean the function doing query/deserialize/update
It's not trivial to include more stuff in the `PollShowResultsEvent` - the other events are created from a network call that includes those in the response from the server
I could pass the whole status into the `StatusActionListener` callback I guess, but even then I'm not sure I always have it

Reply:
Ah I see. Its fine then.